### PR TITLE
fix: [select] label can be displayed correctly after selecting operation

### LINKED ISSUE #8544: [Component] ElSelect showing wrong label
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.36`
- Element Plus Version: `2.2.6`
- Browser / OS: `UserAgent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/102.0.0.0 Safari/537.36`
- Build Tool: `Webpack`

### Related Component

- `el-select`
- `el-option-group`
- `el-option`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbmltcG9ydCB7IHJlZiwgdmVyc2lvbiBhcyB2dWVWZXJzaW9uLCBjb21wdXRlZCwgcmVhY3RpdmUgfSBmcm9tICd2dWUnXG5pbXBvcnQgeyB2ZXJzaW9uIGFzIEVwVmVyc2lvbiB9IGZyb20gJ2VsZW1lbnQtcGx1cydcbmltcG9ydCB7IEVsZW1lbnRQbHVzIH0gZnJvbSAnQGVsZW1lbnQtcGx1cy9pY29ucy12dWUnXG5cbmNvbnN0IG1zZyA9IHJlZignSGVsbG8gV29ybGQhJylcblxuY29uc3QgdmFsdWUgPSByZWYoMClcblxuY29uc3QgZGF0YSA9IHtcbiAgXCJhXCI6IFtcbiAgICBbXG4gICAgICBcImdyb3VwMVwiLFxuICAgICAgW1xuICAgICAgICB7IGluZGV4OiAwLCBuYW1lOiBcInhcIiB9LFxuICAgICAgICB7IGluZGV4OiAxLCBuYW1lOiBcInlcIiB9LFxuICAgICAgICB7IGluZGV4OiAyLCBuYW1lOiBcInpcIiB9LFxuICAgICAgXVxuICAgIF0sXG4gICAgW1xuICAgICAgXCJncm91cDJcIixcbiAgICAgIFtcbiAgICAgICAgeyBpbmRleDogMywgbmFtZTogXCJqXCIgfSxcbiAgICAgICAgeyBpbmRleDogNCwgbmFtZTogXCJrXCIgfSxcbiAgICAgICAgeyBpbmRleDogNSwgbmFtZTogXCJsXCIgfSxcbiAgICAgIF1cbiAgICBdXG4gIF0sXG5cbiAgXCJiXCI6IFtcbiAgICBbXG4gICAgICBcImdyb3VwM1wiLFxuICAgICAgW1xuICAgICAgICB7IGluZGV4OiAwLCBuYW1lOiBcInhcIiB9LFxuICAgICAgICB7IGluZGV4OiAxLCBuYW1lOiBcInlcIiB9LFxuICAgICAgICB7IGluZGV4OiAyLCBuYW1lOiBcInpcIiB9LFxuICAgICAgXVxuICAgIF0sXG4gICAgW1xuICAgICAgXCJncm91cDRcIixcbiAgICAgIFtcbiAgICAgICAgeyBpbmRleDogMywgbmFtZTogXCJqXCIgfSxcbiAgICAgICAgeyBpbmRleDogNCwgbmFtZTogXCJrXCIgfSxcbiAgICAgICAgeyBpbmRleDogNSwgbmFtZTogXCJsXCIgfSxcbiAgICAgICAgeyBpbmRleDogNiwgbmFtZTogXCJsXCIgfSxcbiAgICAgIF1cbiAgICBdXG4gIF1cbn1cblxuY29uc3QgaXRlbSA9IHJlZihcImFcIilcblxuY29uc3QgbWFwID0gY29tcHV0ZWQoKCkgPT4ge1xuICByZXR1cm4gZGF0YVtpdGVtLnZhbHVlXVxufSlcblxuZnVuY3Rpb24gaGFuZGxlQ2xpY2soKSB7XG4gIGl0ZW0udmFsdWUgPSBcImJcIlxuICAvLyB2YWx1ZS52YWx1ZSA9IDBcbn1cbjwvc2NyaXB0PlxuXG48dGVtcGxhdGU+XG4gIDxwPlxuICAgIDxlbC1pY29uIGNvbG9yPVwidmFyKC0tZWwtY29sb3ItcHJpbWFyeSlcIj48RWxlbWVudFBsdXMgLz48L2VsLWljb24+XG4gICAgRWxlbWVudCBQbHVzIHt7IEVwVmVyc2lvbiB9fSArIFZ1ZSB7eyB2dWVWZXJzaW9uIH19XG4gIDwvcD5cbiAgXG4gIDxlbC1zZWxlY3Qgdi1tb2RlbD1cInZhbHVlXCI+XG4gICAgPGVsLW9wdGlvbi1ncm91cFxuICAgICAgdi1mb3I9XCIoZ3JvdXApIGluIG1hcFwiXG4gICAgICA6a2V5PVwiZ3JvdXBbMF1cIlxuICAgICAgOmxhYmVsPVwiZ3JvdXBbMF1cIlxuICAgID5cbiAgICAgIDxlbC1vcHRpb25cbiAgICAgICAgdi1mb3I9XCJpdGVtIGluIGdyb3VwWzFdXCJcbiAgICAgICAgOmtleT1cIml0ZW0ubmFtZVwiXG4gICAgICAgIDpsYWJlbD1cIml0ZW0ubmFtZVwiXG4gICAgICAgIDp2YWx1ZT1cIml0ZW0uaW5kZXhcIlxuICAgICAgPjwvZWwtb3B0aW9uPlxuICAgIDwvZWwtb3B0aW9uLWdyb3VwPlxuICA8L2VsLXNlbGVjdD5cblxuICA8ZWwtYnV0dG9uIEBjbGljaz1cImhhbmRsZUNsaWNrXCI+Y2xpY2s8L2VsLWJ1dHRvbj5cbiAgXG4gIDxwPlxuICAgIGNsaWNrIHRoZSBidXR0b24gYW5kIHNlbGVjdCBvdGhlciBvcHRpb25zLCBpdCB3aWxsIG5vdCBzaG93aW5nIHRoZSBsYWJlbCwgYnV0IHNob3dpbmcgdGhlIHZhbHVlXG4gIDwvcD5cbjwvdGVtcGxhdGU+XG4iLCJpbXBvcnRfbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7fVxufSIsIl9vIjp7fX0=)

### Steps to reproduce

see the playground

### What is Expected?

The ElSelect should correctly showing the option label, not option value

### What is actually happening?

When options are changed, ElSelect is showing the option value instead of option value

### Additional comments

_(empty)_

### PR DESCRIPTION
**Pull request by baiwusanyu-c**
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

close: #8544 
When props are changed, label can be displayed correctly after selecting operation

### CODE REVIEW

**Comment by baiwusanyu-c** [comment]
> Please make sure these boxes are checked before submitting your PR, thank you!
> 
> * [x]  Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
> * [x]  Make sure you are merging your commits to `dev` branch.
> * [x]  Add some descriptions and refer to relative issues for your PR.
> 
> close: #8544 When props are changed, label can be displayed correctly after selecting operation

**Comment by wangXiaoShow** [comment]
Who can review this PR ? @tolking

File: packages/components/select/src/option.vue [hunk 1]
```diff
@@ -67,7 +67,15 @@ export default defineComponent({
...
-
+    // fix: 8544
+    //  When props are changed, the component is remounted,
+    //  and "onOptionCreate" will be called,
+    //  but in the previous hook method "onBeforeUnmount",
+    // "cachedOptions.delete" will be executed after "onOptionCreate", which leads to a bug (8544).
+    // Here, call "onOptionCreate" again
+    nextTick(() => {
+      select.onOptionCreate(vm as unknown as SelectOptionProxy)
+    })
```
Comment (tolking) [inline packages/components/select/src/option.vue:78]: I do not think this is the best way, too many nextTick will make the component difficult to maintain.

I think the problem is in line 78, the if should be inside nextTick.

Comment (tolking) [inline packages/components/select/src/option.vue:78]: I do not think this is the best way, too many nextTick will make the component difficult to maintain.

I think the problem is in line 78, the if should be inside nextTick.

Comment (baiwusanyu-c) [reply packages/components/select/src/option.vue]: You're right ！

File: packages/components/select/__tests__/select.test.ts [hunk 2]
```diff
@@ -161,7 +161,11 @@ const getSelectVm = (configs: SelectProps = {}, options?) => {
   )
 }
 
-const getGroupSelectVm = (configs: SelectProps = {}, options?) => {
+const getGroupSelectVm = (
+  configs: SelectProps = {},
+  options?,
+  groupTemplate?
+) => {
```
Comment (tolking) [inline packages/components/select/__tests__/select.test.ts:168]: This seems redundant, you should keep the data format consistent with options, and modify the options through `wrapper.vm.options`.

Comment (tolking) [inline packages/components/select/__tests__/select.test.ts:168]: This seems redundant, you should keep the data format consistent with options, and modify the options through `wrapper.vm.options`.

File: packages/components/select/src/option.vue [hunk 3]
```diff
@@ -67,19 +67,18 @@ export default defineComponent({
     const vm = getCurrentInstance().proxy
     const key = (vm as unknown as SelectOptionProxy).value
     select.onOptionCreate(vm as unknown as SelectOptionProxy)
-
```
Comment (tolking) [inline packages/components/select/src/option.vue:70]: This newline can be preserved

File: packages/components/select/__tests__/select.test.ts (hunk 2 above)
Comment (baiwusanyu-c) [reply packages/components/select/__tests__/select.test.ts]: I have modified it

**Review by tolking — APPROVED** [review]

**Comment by baiwusanyu-c** [comment]
😳

**Pull request merged**